### PR TITLE
RDFJS Source interface on N3Store

### PR DESCRIPTION
**Pull request by iddan**
[Source](http://rdf.js.org/#source-interface) interface for Store implementation
Based on https://github.com/rdfjs/N3.js/pull/161

### CODE REVIEW

**Comment by iddan** [comment]
Ready for review

**Comment by RubenVerborgh** [comment]
Rebased on latest master, will review now.

File: lib/N3Stream.js [hunk 1]
```diff
@@ -0,0 +1,5 @@
+var Stream = require('stream').Stream;
+
+class N3Stream extends Stream {}
+
+module.exports = N3Stream;
```
Comment (RubenVerborgh) [inline lib/N3Stream.js]: This class is trivial; should not be included.

File: test/N3Store-test.js [hunk 2]
```diff
@@ -1036,6 +1037,38 @@ describe('N3Store', function () {
...
+      var stream = store.match();
+      var data = [];
+      stream.on('data', (quad) => {
+        data.push(quad);
+      });
+      store.addQuad('null', 'null', 'null', 'null');
+      data.should.deep.equal([new Quad('null', 'null', 'null', 'null')]);
+    });
+
+    it('should emit quad every time a matching quad is added to the store', function () {
```
Comment (RubenVerborgh) [inline test/N3Store-test.js]: That's not how I understand the `Source` interface. Is it stated somewhere that the stream should keep on updating indefinitely?

File: lib/N3Store.js [hunk 3]
```diff
@@ -19,6 +21,8 @@ class N3Store {
     this._entities = Object.create(null); // inverse of `_ids`
     // `_blankNodeIndex` is the index of the last automatically named blank node
     this._blankNodeIndex = 0;
+    // '_changeStream' holds active streams created by this.match()
+    this._changeStream = new N3Stream();
```
Comment (RubenVerborgh) [inline lib/N3Store.js]: Not sure about the necessity; see below.

File: test/N3Store-test.js (hunk 2 above)
Comment (iddan) [reply test/N3Store-test.js]: I guess you can interpret it like that. In RDFLib you can listen to additions and removals of the store. If that's not the intended API I can change it.

**Comment by RubenVerborgh** [comment]
I don’t think its the intended API; listening to changes would be a different mechanism.

**Comment by iddan** [comment]
I will propose a different API.

**Pull request closed without merging**